canu/utils/sls.py

```diff
@@ -119,7 +119,7 @@ def remote_request(
         sls_cache = None
 
         if os.getenv("SLS_API_GW") is None:
-            sls_url = "http://cray-sls.services.svc.cluster.local/v1/networks"
+            sls_url = "https://cray-sls.services.svc.cluster.local/v1/networks"
```

Comment:
this won't work inside the mesh:

```
ncn-m001:~ # kubectl exec -it -n services cray-dhcp-kea-5c7f885c9c-hcn5h -c cray-dhcp-kea -- /bin/bash
bash-5.1$ curl "https://cray-sls.services.svc.cluster.local/v1/networks"
curl: (35) OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to cray-sls.services.svc.cluster.local:443 
bash-5.1$ curl "http://cray-sls.services.svc.cluster.local/v1/networks"
[{"Name":"NMN_RVR","FullName":"River Compute Node Management Network","IPRanges
```

Reply:
add a comment that the "if" side of this is in-mesh and the else is out of mesh because this is confusing from before your change.

Reply:
So add a command regarding in/out-mesh, but revert this to http and silence Sonarcloud?

Reply:
Fixed